attractions/dropdown/dropdown.svelte

```diff
@@ -13,9 +22,78 @@
    * @type {boolean}
    */
   export let top = false;
+  /**
+   * Dropdown horizontal alignment relative to the shell.
+   * @type {string}
+   */
+  export let horizontalAlignment = HorizontalAlignment.AUTO_START;
+  /**
+   * Dropdown vertical alignment relative to the shell.
+   * @type {string}
+   */
+  export let verticalAlignment = VerticalAlignment.AUTO_BOTTOM;
+
+  const isDropdownOpen = getContext(CONTEXT_IS_DROPDOWN_OPEN);
+  const getDropdownShellBoundary = getContext(
+    CONTEXT_GET_DROPDOWN_SHELL_BOUNDARY
+  );
+
+  let dropdownElement, isVerticalAlignTop, isHorizontalAlignEnd;
+
+  function getIsVerticalAlignTop(dropdownBound, dropdownShellBound) {
+    const { height } = dropdownBound;
+    const { top, bottom } = dropdownShellBound;
+
+    switch (verticalAlignment) {
+      case VerticalAlignment.TOP:
+        return true;
+      case VerticalAlignment.BOTTOM:
+        return false;
+      case VerticalAlignment.AUTO_TOP:
+        return height <= top;
+      default:
+        // auto-bottom by default
+        return height > window.innerHeight - bottom;
+    }
+  }
+
+  function getIsHorizontalAlignEnd(dropdownBound, dropdownShellBound) {
+    const { width } = dropdownBound;
+    const { left } = dropdownShellBound;
+
+    switch (horizontalAlignment) {
+      case HorizontalAlignment.END:
+        return true;
+      case HorizontalAlignment.START:
+        return false;
+      case HorizontalAlignment.AUTO_END:
+        return width <= left;
+      default:
+        // auto-start by default
+        return width > window.innerWidth - left;
```

Comment:
Now I'm also contemplating if we should allow specifying which element to use as the boundary. For instance, if we have an element with `overflow: hidden`, we'd want to position the dropdown within the bounds of that element. Can we do that by just passing a DOM node of the respective element, defaulting to `window`?

Reply:
Sorry, I was very busy in the previous months.

For this feature, It is quite an edge case and I'm not sure if it will be worth introducing the complexity to the code base. **window** use `innerWidth` and `innerHeight` while **HTMLElement** use `clientHeight` and `clientWidth`. And when we use HTMLElement, which is mostly smaller than the window, then its position must be included in the calculation as well.

Reply:
Alright, we might add it in the future if there will be any requests for it.